src/firmware/guest/types/snp.rs

```diff
@@ -282,42 +282,122 @@ pub struct AttestationReport {
 }
 
 impl AttestationReport {
+    #[inline(always)]
+    /// This will never panic because we are enforcing the length of
+    /// the slice before checking.
+    fn chip_id_is_turin_like(bytes: &[u8]) -> bool {
+        // Chip ID -> 0x1A0-0x1E0
+        // Last 8 bytes of CHIP_ID are zero, then it is Turin Like.
+        bytes == [0; 56]
+    }
+
+    #[inline(always)]
+    /// Check if MASK_CHIP_ID is set before checking chip_id_is_turin_like
+    fn mask_chip_id_en(bytes: &[u8]) -> bool {
```

Comment:
The `MASK_CHIP_ID` is never checked in this method? It only checks if the first 4 bytes of some array?

Reply:
It's sole purpose is to check the second bit of a 32 bit value. Where the method is private, and we are specifically calling it by passing the bytes from the offset specified in the SEV-SNP ABI, it is why I named it and commented as such.

Reply:
Would you prefer the following, instead:

```rust
fn mask_chip_id_en(bytes: &[u8]) -> bool {
    let mut byte_array: [u8; 4] = [0; 4];
    byte_array.copy_from_slice(bytes);
    KeyInfo(u32::from_bytes(byte_array)).mask_chip_key()
}
 ```

Reply:
NVM, we can't use the `KeyInfo` fields from the AttestationReport, as they do not include mask_chip_id, but mask_chip_key, and that doesn't indicate if mask_chip_id is enabled or not. Removed that check.